### ISSUE TITLE
Github.io comments: Adding colors to your visual

### ISSUE THREAD

**Issue by uve** [microsoft/PowerBI-visuals#432]
Comments to https://microsoft.github.io/PowerBI-visuals/docs/how-to-guide/adding-colors-to-your-visual/

**Comment by mattroot87** [comment]
@uve Are all the visuals pulling from the same host.colorPalette object?  If not how do I sync the colors assigned to the data points in my host.colorPalette with the data point colors from other visuals?

Thank you!

Matt

**Comment by helen508light** [comment]
@mattroot87 
it is not impossible to manage another visual's state from outside(from another visual)
Any visual receive theme/colorPalette only for itself

**Comment by mattroot87** [comment]
Thank you @helen508light 

If I am understanding you correctly what you are saying is that if I am using the colorPalette to assign colors to myy data points in my custom visual there is no way to guarantee that the colors from other visuals are going to assign the same colors to the data points as my visual?  Thank you.

**Comment by helen508light** [comment]
yep

**Comment by philip-garner-tessella** [comment]
@mattroot87, did you find any workarounds for this?

Not being able to use consistent colors across visuals is proving to be a bit of a pain.

**Comment by mattroot87** [comment]
In order to get consistent colors across visuals I have found the ‘measure’ has to be the same column. Let me know if that makes sense or you need more info.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by twastvedt** [comment]
@mattroot87 Were you able to get consistent colors between a custom visual and a built-in visual? For example, I have a (built-in) pie chart who's legend uses the same column as my custom visual, but `getColor(category.values[i])` returns different colors for each category than are shown in the pie chart. Thanks!

**Comment by mattroot87** [comment]
When I asked the Power BI Custom Visual folks they told me there isn't a guaranteed way to get consistant colors across custom visuals and out of the box visuals.  That being said, I noticed that if I have the same measure column set in my custom visual and in the out of the box visual I could get accurate results.

**Comment by twastvedt** [comment]
Cool, thanks for the reply! Trying to follow in your footsteps, but I wasn't able to duplicate that result. I imagine I have something different in my setup. I also noticed that, starting from a blank page, the colors assigned to a particular column value change depending on the fields used in the first visual created. So, I don't know how the custom visual would know how to match up values, unless the sorting of the values in the dataView happens to be the same as the sorting used to create that first visual. I also don't know what determines the sorting of the values in the dataView.

**Comment by mattroot87** [comment]
Do you have a measure column in your custom visual?

**Comment by twastvedt** [comment]
Not sure what you mean by "measure column". I have columns. No measures. The color I'm trying to match is tied to a column in the data. In "dataViewMappings" the data role is of kind "GroupingOrMeasure".

**Comment by mattroot87** [comment]
Sorry for my bad terminology...

What I did was I had two data fields that were mapped to data roles.  One for grouping and the other for measure. Just like this...

![image](https://user-images.githubusercontent.com/24396979/78993568-f1372080-7b0b-11ea-9c9f-366f5ac01381.png)

What I noticed was that the color palette assigns the color to the data points ('grouping'  data role) based on what field you pull into the measure data field- for the above example that would be  "My Measure Data" field.  If your Measure data field in your custom visual has the same column from your data source as you pull into the Measure data field in the other out of the box visuals, than the color pallete from the out of the box visual will assign the same colors as your color pallete will assign in your custom visual. I couldn't get that guaranteed by the Power BI Custom visual folks but it has been working ok for me so far.

Hopefully that helps more.
